vars/piperStageWrapper.groovy

```diff
@@ -99,7 +99,11 @@ private void executeStage(script, originalStage, stageName, config, utils) {
     } finally {
         //Perform stashing of selected files in workspace
         utils.stashList(script, script.commonPipelineEnvironment.configuration.stageStashes?.get(stageName)?.stashes ?: [])
-        deleteDir()
+        //NOTE: We do not delete the directory because it failed on JaaS because of infrastructure issues.
+        // deleteDir() is not required in pods, but would be nice to have the same behaviour and leave a clean fileSystem.
+        if (isNotInsidePod(script)) {
```

Comment:
Isn't it unusual to have a method checking an "isNot". Wouldn't it more usual to have a "is" and to negate with a `!`.
With other words:

`if(! isBlah())` rather than `if(isNotBlah())`

Reply:
Fixed in the same follow-up commit.